### PR TITLE
[FIX] sale_ux: update_prices fixed

### PR DESCRIPTION
**Pull request by vib-adhoc**

### CODE REVIEW

**Comment by roboadhoc** [comment]
[Pull request status dashboard](https://runbot.adhoc.com.ar/ingadhoc/sale/pull/575).

File: sale_ux/models/sale_order.py [hunk 1]
```diff
@@ -104,14 +104,19 @@ def _get_forbidden_state_confirm(self):
...
     def update_prices(self):
         # for compatibility with product_pack module
         pack_installed = 'pack_parent_line_id' in self.order_line._fields
         if pack_installed:
             products_pack = self.order_line.with_context(update_prices=True, pricelist=self.pricelist_id.id).filtered(
-                pack_installed and l.product_id.pack_ok and l.product_id.pack_component_price != 'ignored')
-            super(SaleOrder, self - products_pack).update_prices()
-            lines_to_update = []
+                lambda l: l.product_id.pack_ok and l.product_id.pack_component_price != 'ignored')
+            super(SaleOrder, self.with_context(lines_to_not_update_ids=products_pack.mapped('id'))).update_prices()
```
Comment (nicomacr) [inline sale_ux/models/sale_order.py]:
```suggestion
            super(SaleOrder, self.with_context(lines_to_not_update_ids=pack_lines.ids)).update_prices()
```

File: sale_ux/models/sale_order.py [hunk 2]
```diff
@@ -104,14 +104,19 @@ def _get_forbidden_state_confirm(self):
         # 'draft' or 'sent'
         return super()._get_forbidden_state_confirm() | set({'sale'})
 
+    def _get_update_prices_lines(self):
+        res = super()._get_update_prices_lines()
+        lines_to_not_update_ids = self._context.get('lines_to_not_update_ids', False)
+        res = res.search([('id', 'not in', lines_to_not_update_ids),('id', 'in', res.mapped('id'))])
+        return res
```
Comment (nicomacr) [inline sale_ux/models/sale_order.py]:
```suggestion
        lines = super()._get_update_prices_lines()
        lines_to_not_update_ids = self._context.get('lines_to_not_update_ids', [])
        return lines.filtered(lambda line: line.id not in lines_to_not_update_ids)
```

File: sale_ux/models/sale_order.py [hunk 3]
```diff
@@ -104,14 +104,19 @@ def _get_forbidden_state_confirm(self):
...
+
     def update_prices(self):
         # for compatibility with product_pack module
         pack_installed = 'pack_parent_line_id' in self.order_line._fields
         if pack_installed:
             products_pack = self.order_line.with_context(update_prices=True, pricelist=self.pricelist_id.id).filtered(
```
Comment (nicomacr) [inline sale_ux/models/sale_order.py]:
```suggestion
            pack_lines = self.order_line.with_context(update_prices=True, pricelist=self.pricelist_id.id).filtered(
```

File: sale_ux/models/sale_order.py [hunk 4]
```diff
@@ -104,14 +104,19 @@ def _get_forbidden_state_confirm(self):
...
-                pack_installed and l.product_id.pack_ok and l.product_id.pack_component_price != 'ignored')
-            super(SaleOrder, self - products_pack).update_prices()
-            lines_to_update = []
+                lambda l: l.product_id.pack_ok and l.product_id.pack_component_price != 'ignored')
+            super(SaleOrder, self.with_context(lines_to_not_update_ids=products_pack.mapped('id'))).update_prices()
             for line in products_pack:
```
Comment (nicomacr) [inline sale_ux/models/sale_order.py]:
```suggestion
            for line in pack_lines:
```

**Review by nicomacr — CHANGES_REQUESTED** [review]

**Review by nicomacr — APPROVED** [review]

**Comment by nicomacr** [comment]
@roboadhoc r+

**Comment by nicomacr** [comment]
@roboadhoc override=ci/oba_without_tests p=0

**Pull request closed without merging**